crates/libs/bindgen/src/lib.rs

```diff
@@ -109,12 +109,23 @@ pub fn namespace(gen: &Gen, tree: &Tree) -> String {
 
     let types = types.values();
 
-    let tokens = quote! {
+    let mut tokens = quote! {
         #(#namespaces)*
         #(#functions)*
         #(#types)*
     };
 
+    if tree.namespace == "Windows.Win32.UI.WindowsAndMessaging" {
+        tokens.combine(&quote! {
+            #[cfg(target_pointer_width = "32")]
+            #[cfg(feature = "Win32_Foundation")]
+            pub use SetWindowLongA as SetWindowLongPtrA;
+            #[cfg(target_pointer_width = "32")]
+            #[cfg(feature = "Win32_Foundation")]
+            pub use GetWindowLongA as GetWindowLongPtrA;
```

Comment:
This seems to be missing the Unicode versions (`SetWindowLongPtrW` and `GetWindowLongPtrW`). Is this intentional or an oversight?

There's also a slight discrepancy between types: For 32-bit targets the functions accept/return `i32`, for 64-bit targets `isize`. I wonder if that has any practical implications for clients. Current stable Rust (1.62.1) doesn't warn about superfluous casts (such as `0_isize as isize`), so I'm guessing this is ok.

Reply:
I left out the W versions since the A and W versions are identical under the hood - there's really no reason to have both functions. Another less confusing approach might be to consolidate them by removing the W versions and dropping the suffix from the remaining version. Then you'd effectively get the post-macro names from C++ e.g. `SetWindowLong` and `GetWindowLong`. 

Yes, I wondered about the return type as well but all the samples I ported handled it naturally so this seems to work fine in practice.

Reply:
I don't feel that this type of consolidation should be done here. That's really something the Platform/SDK Team would need to initiate.

Whether the implementations are identical or not is not something us developers can observe. We have to go by the [documentation](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowlongptra) and as long as that documentation explicitly says: "Don't mix encodings" then that's what those of us that program against a documented contract will have to do.

I'm not opposed to keeping things as simple as possible. I just feel that attempting to fix this here is going to lead to a more confusing situation.

Reply:
That's fair. In that case, I'm inclined to do nothing and just let this be handled downstream, perhaps by a dedicated crate.

Reply:
I hear ya but getting the platform to fix this will be impossible. And creating/consuming a separate crate to fix this special case is not ideal. I think we should keep these as ugly as they are.

Reply:
> I think we need to add the W variants tho.

Agreed. As far as I understand, the ANSI and Unicode versions are distinct exports, with distinct ordinals and distinct addresses. Leaving either one out is harmful to clients that need the function addresses for reasons other than calling them. Not an issue with `windows` (which doesn't expose the raw entry points), but for `windows-sys` it is, and it would just feel awkward to not have those around for reasons we cannot reasonably verify.

Reply:
I'll refrain from discussing the implementation details, but agree that it would be more consistent to provide both. Coming up.